File: database/database-annotation-processor/src/test/java/io/koraframework/database/common/annotation/processor/repository/error/UnknownQueryParameter.java
package io.koraframework.database.common.annotation.processor.repository.error;

import io.koraframework.database.common.annotation.Query;
import io.koraframework.database.common.annotation.Repository;
import io.koraframework.database.jdbc.JdbcRepository;

@Repository
public interface UnknownQueryParameter extends JdbcRepository {

    @Query("SELECT * FROM table WHERE id = :userId")
    String wrongParameterName(long id);
}

File: database/database-symbol-processor/src/test/kotlin/io/koraframework/database/symbol/processor/repository/error/UnknownQueryParameter.kt
package io.koraframework.database.symbol.processor.repository.error

import io.koraframework.database.common.annotation.Query
import io.koraframework.database.common.annotation.Repository
import io.koraframework.database.jdbc.JdbcRepository

@Repository
interface UnknownQueryParameter : JdbcRepository {

    @Query("SELECT * FROM table WHERE id = :userId")
    fun wrongParameterName(id: Long): String
}
